gestalt-asset-core/src/main/java/org/terasology/gestalt/assets/AssetType.java

```diff
@@ -153,23 +157,23 @@ public synchronized boolean isClosed() {
      * Disposes all assets of this type.
      */
     public synchronized void disposeAll() {
-        loadedAssets.values().forEach(T::dispose);
-
-        for (WeakReference<T> assetRef : ImmutableList.copyOf(instanceAssets.values())) {
-            T asset = assetRef.get();
-            if (asset != null) {
+        loadedAssets.values().forEach(k -> {
+            Asset<U> asset = k.get();
+            if(asset != null) {
```

Comment:
Need a space between the if and (
:P

Reply:
It probably isn't impossible for the parent asset to be disposed while instance assets exist - this check may leave those undisposed.